asserts/model.go

```diff
@@ -25,16 +25,310 @@ import (
 	"strings"
 	"time"
 
+	"github.com/snapcore/snapd/snap/channel"
 	"github.com/snapcore/snapd/snap/naming"
 	"github.com/snapcore/snapd/strutil"
 )
 
+// ModelSnap holds the details about a snap specified by a model assertion.
+type ModelSnap struct {
+	Name   string
+	SnapID string
+	// SnapType is one of: app|base|gadget|kernel|core, default is app
+	SnapType string
+	// Modes in which the snap must be made available
+	Modes []string
+	// DefaultChannel is the initial tracking channel, default is stable
+	DefaultChannel string
+	// Track is a locked track for the snap, if set DefaultChannel cannot be
+	Track string
+	// Presence is one of: required|optional
+	Presence string
+}
+
+// SnapName implements naming.SnapRef.
+func (s *ModelSnap) SnapName() string {
+	return s.Name
+}
+
+// ID implements naming.SnapRef.
+func (s *ModelSnap) ID() string {
+	return s.SnapID
+}
+
+type modelSnaps struct {
+	Base   *ModelSnap
+	Gadget *ModelSnap
+	Kernel *ModelSnap
+	Snaps  []*ModelSnap
```

Comment:
`Snaps` doesn't indicate if it contains Base, Gadget and Kernel. Perhaps a comment or a tweak to naming (e.g. calling this field `Others` would alleviate that?)

Reply:
s/Snaps/SnapsNoEssential/, also not sure why I used exported names here